src/partial-assembly.md

```diff
@@ -0,0 +1,171 @@
+`OperatorJacobiSmoother`. This class efficiently assembles the diagonal of the
+corresponding matrix, exploiting the tensor-product structure for efficient
+evaluation.
+
+MFEM also allows for [Chebyshev
+smoothing](http://netlib.org/linalg/html_templates/node76.html) with partial
+assembly using the class `OperatorChebyshevSmoother`. This smoother uses
+estimates of the eigenvalues of the operator computed using the [power
+method](https://en.wikipedia.org/wiki/Power_iteration), and is built upon the
+functionality of `OperatorJacobiSmoother`.
+
+These smoothers can be used within the context of h- and p-multigrid methods.
+These facilities are provided using the `Multigrid` class. This functionality is
+illustrated in [example 26](examples.md#ex26).
+
+
+## Finite Element Operator Decomposition
+
+The partial assembly functionality in MFEM is based on decomposing the finite
+element operator into a nested sequence of operations that act on different
+levels of the discretization.
+
+Finite element operators are typically defined through weak formulations of
+partial differential equations that involve integration over a computational
+mesh. The required integrals are computed by splitting them as a sum over the
+mesh elements, mapping each element to a simple *reference* element (e.g. the
+unit square) and applying a quadrature rule in reference space.
+
+This sequence of operations highlights an inherent hierarchical structure
+present in all finite element operators where the evaluation starts on *global
+(trial) degrees of freedom (dofs) on the whole mesh*, restricts to *degrees of
+freedom on subdomains* (groups of elements), then moves to independent *degrees
+of freedom on each element*, transitions to independent *quadrature points* in
+reference space, performs the integration, and then goes back in reverse order
+to global (test) degrees of freedom on the whole mesh.
+
+This is illustrated below for the case of a symmetric linear operator. We use
+the notions **T-vector**, **L-vector**, **E-vector** and **Q-vector** to
+represent the sets corresponding to the (true) degrees of freedom on the global
+mesh, the split local degrees of freedom on the subdomains, the split degrees
+of freedom on the mesh elements, and the values at quadrature points,
+respectively.
+
+We refer to the operators that connect the different types of vectors as:
+
+- Subdomain restriction **P**
+- Element restriction **G**
+- Basis (Dofs-to-Qpts) evaluator **B**
+- Operator at quadrature points **D**
+
+More generally, when the test and trial space differ, they get their own
+versions of **P**, **G** and **B**.
+
+![Operator Decomposition](img/libceed.png "Operator Decomposition")
+
+Note that in the case of adaptive mesh refinement (AMR), the restrictions **P**
+and **G** will involve not just extracting sub-vectors, but evaluating values
+at constrained degrees of freedom through the AMR interpolation. There can also
```

Comment:
Small correction: actually, currently in MFEM, nonconforming AMR is handled exclusively by the prolongation operator `P`.

Awesome documentation, by the way. 👍